### DISCUSSION TITLE
Get Access to java JNI wrapper code.

### DISCUSSION THREAD

**Discussion by ByteGeek** [kuzudb/kuzu#2746] [Q&A] [answered]
Hi, we are trying to setup Kuzu DB within the context of an android application. Our system uses API level 25 which does not support File/NIO classes which seem to be used in the wrapper.

Can we get access to the Java wrapper library as a repository in GIT? Would prefer not to decompile the JAR and work on it.

**Comment by mewim** [comment, marked as answer]
Hi @ByteGeek,

The source code of our Java bindings is under https://github.com/kuzudb/kuzu/tree/master/tools/java_api

**Comment by ByteGeek** [comment]
Thank you, was able to modify the code and get it to run.
Dynamically, the .so files tries to link against libpthread.so.0 which android does not support. 
In android, threading is handled by libc.so, can we get a version of the library that does not link against pthread?

**Comment by mewim** [comment]
@ByteGeek I think the problem might not be limited to just pthread. Our binary is pre-compiled for Linux with glibc. I think Android uses Bionic libc instead, which is not compatible with glibc. To make the binary work on Android, the binary needs to be built by linking to Bionic.
